### DISCUSSION TITLE
SPINDELMATE ver1.4.0

### DISCUSSION THREAD

**Discussion by kbaggen** [kbaggen/SPINDELMATE#10] [Announcements]
SPINDELMATE version1.4.0

- Updated code  for PILL identification as KEGLAND changed the RAPT Pill telemetric data, and hence, need new way to identify RAPT Pill´s. Therefore, please use "100" at "TILT Color/RAPT Pill/iSPINDEL" at setting. This find all RAPT Pills and hence if you have more than one you need to get serial output on to get the mac:id and use mac:id insteed, e.g. 78:e4:8d:4d:21:4c. You can use Brewflasher, ESPhome flasher or Arduino IDE to show the serial output from used com-port and just await it list the found Rapt Pill after bluetooth scanning is completed. 
- AP mode for iSPINDEL has been removed as having SPINDELMATE in both AP (access point mode) and STA mode is giving instability issues. Sorry, metal fermenters users. 
- Stability code enhancements.

<hr /><em>This discussion was created from the release <a href='https://github.com/kbaggen/SPINDELMATE/releases/tag/'>SPINDELMATE ver1.4.0</a>.</em>

**Comment by ESEMCE** [comment]
I don't know if it's my hardware or if there is a software issue,
But I am getting the following error on both 1.4.0 and 1.5.0.

The ESP32 then reboots.

[19:19:53]Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.
[19:19:53]Core 1 register dump:
[19:19:53]PC      : 0x400013f9  PS      : 0x00060e30  A0      : 0x800d4d62  A1      : 0x3ffcb9e0  
[19:19:53]A2      : 0x3ffc2700  A3      : 0x00000000  A4      : 0x000000ff  A5      : 0x0000ff00  
[19:19:53]A6      : 0x00ff0000  A7      : 0xff000000  A8      : 0x00000000  A9      : 0x3ffcb9b0  
[19:19:53]A10     : 0x3ffc2700  A11     : 0x3f4006c5  A12     : 0x00000000  A13     : 0x00000001  
[19:19:53]A14     : 0x00000000  A15     : 0x3ffcbbbc  SAR     : 0x0000000a  EXCCAUSE: 0x0000001c  
[19:19:53]EXCVADDR: 0x00000000  LBEG    : 0x400013f9  LEND    : 0x4000140d  LCOUNT  : 0xffffffff  
[19:19:53]
[19:19:53]ELF file SHA256: 0000000000000000
[19:19:53]
[19:19:53]Backtrace: 0x400013f9:0x3ffcb9e0 0x400d4d5f:0x3ffcb9f0 0x400d7d15:0x3ffcbbf0 0x401037ba:0x3ffcc0f0 0x40091092:0x3ffcc110

**Reply by kbaggen** [reply]
Can you show a picture or link of the board?

**Reply by ESEMCE** [reply]
It's the 30 pin micro USB from this listing
https://a.aliexpress.com/_m02eCMS

**Reply by kbaggen** [reply]
Try take a look on the Chip, and see if it states "ESP-WROOM-32". If it state "ESP-WROOM-32**U**" or "ESP-WROOM-32**D**" as the listing you link tilll also seem to sell according till the pictures, I assume this is the reason. The code is build to support ESP-WROOM-32, but it seems some of the newer chips revisions might have issues........

**Reply by ESEMCE** [reply]
It only indicates ESP-WROOM-32 with no suffix and has "ESP32 Devkit v1" printed on the back of the PCB like your photos.

But looking at the FCC ID, it looks like there might be a difference between 2AC7Z-ESPWROOM32 and 2AC7Z-ESP _32_ WROOM32 . I can't find any supplier that doesn't use ESP32WROOM32 from those that have photos of boards clear enough to read.
Can you recommend a reliable vendor?

![MVIMG_20230815_071900](https://github.com/kbaggen/SPINDELMATE/assets/57524932/fddfdffb-ab35-4fd8-b946-22588908cd06)

**Comment by kbaggen** [comment]
Hi agian, from what I can see is should be the correct board, but there is tons of various EPS32 out there and not all vendors wirte the right stuff on them assumingly. But it states WSP-WROOM-32, and hold the pin layout. Hence, it might be faulty...at least with my code :-)

I pretty sure I brought these one back in time: 
https://jentronic.dk/iot-mainboards/45-doit-esp32-iot-mainboard.html?gclid=CjwKCAjwxOymBhAFEiwAnodBLDHX1sO7DU8zzZiMFkVLewIX_NXpdw_Af89CyRcrO1Ks-bZ554zy4RoCTK0QAvD_BwE

(E.g. DOit ESP32 Devkit1 "ESP-WROOM-32).


I have tested it on ESP32 Devkit1, NodeMCU ESP32S, and then Wemos ESP32 D1, and all is ESP-WROOM-32 or ESP-WROOM-32S!

But things to check: 
- did you remember to hold down "boot" during install as this is needed on most EPS32 to flash it? 
- And else it might run former code you played around with? 
- I see you got something attached to it, and this might also need to be removed as it might also make issues? 
- Can you get it to run other code?
- Try other USB power, at least 5V and 2A?


Hope it helps.

**Reply by ESEMCE** [reply]
Going through your list,
1) Yes
2) the first code was V1.4 of your code using the browser flasher, I've since flashed 1.4 using brewflasher and also flashed v1.5 all crash at the same point as far as I can tell from the logs.
3) That's a LM393 connected
4) I haven't tried, but it does look like your code works to a point.
5) It appears that the same issue occurs when powered by a QC3.0 charger (is there a way to see the logs when not connected to a PC?)

The code seems to run OK, I can see that it connects to my account on https://spindelmate.bubble-logger.com/login.php, downloads settings, then it crashes.

I originally connected the LM393 (just cause it was a solderless option). It crashed with that connected, so I removed it completely but it still crashed. Hoping that it was just a faulty LM393 and the code NEEDS an input to run properly, I just reconnected the LM393 without the VCC connector (to keep this project all together while it moves around the house and soldered on the MPU6050.
I have also run the board with the LM393 completely removed with the same problem.
I can see that it detects the MPU6050 on bootup (I didn't see any mention of the LM393 on bootup) which makes me think things are working to some extent, but the same error occurs after it connects the the server.
I've fiddled around with the server settings but even reverting back to what I recall were the defaults, it still crashes.

I've also tried connecting to my guest WiFi which has a simpler password (my Normal WiFi includes special characters that I thought might be introducing problems)

So it looks like my board is not compatible whether because it's faulty or because it just has some minor difference.

**Comment by kbaggen** [comment]
Ok, connect back to pc and try send some more log so I can see the parts before it go down.

Mighty be an error with settings etc

Secondly, what is your username? (It is in the log if sufficiently long, but just in case)

**Reply by ESEMCE** [reply]
This is the full log, pretty sure I'm running V 1.4

Username: 990xxxx

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0018,len:4
load:0x3fff001c,len:1044
load:0x40078000,len:10044
load:0x40080400,len:5872
entry 0x400806ac

mounting FS...
mounted file system
reading config file
opened config file
{}
parsed json
*wm:[2] Starting Config Portal 
*wm:[2] Disabling STA 
*wm:[2] Enabling AP 
*wm:[1] StartAP with SSID:  SPINDELMATE CONNECT 990xxxx
*wm:[2] AP has anonymous access! 
*wm:[1] AP IP address: 192.168.4.1
*wm:[1] Starting Web Portal 
*wm:[2] HTTP server started 
*wm:[2] Config Portal Running, blocking, waiting for clients... 
*wm:[2] Portal Timeout In 60 seconds
*wm:[2] Portal Timeout In 33 seconds
*wm:[2] Portal Timeout In 3 seconds
*wm:[1] config portal has timed out 
*wm:[2] shutdownConfigPortal 
*wm:[0] [ERROR] disconnect configportal - softAPdisconnect FAILED 
*wm:[2] restoring usermode STA
*wm:[2] wifi status: Unknown
*wm:[2] wifi mode: STA
*wm:[2] configportal closed 
*wm:[1] config portal exiting 
*wm:[1] AutoConnect 
*wm:[2] ESP32 event handler enabled 
*wm:[2] Connecting as wifi client... 
*wm:[2] setSTAConfig static ip not set, skipping 
*wm:[1] Connecting to SAVED AP: WiFi Name
*wm:[1] connectTimeout not set, ESP waitForConnectResult... 
*wm:[2] Connection result: WL_CONNECTED
*wm:[1] AutoConnect: SUCCESS 
*wm:[1] STA IP Address: 192.168.3.85
connected...yeey :)
Web Server started
192.168.3.85
CONNECTED :-)
 Settings data recvied
{"sensitivity":"0","blop_time_in_min":"6","TILTcolor":"0","my_polynominal":"","iblopper4tilt_normal_mode":"","emulate_TILT":"7","ispindel_include":"no","Ispindel_factor_for_TempControl":"1"}
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.
Core 1 register dump:
PC      : 0x400013f9  PS      : 0x00060e30  A0      : 0x800d4b0e  A1      : 0x3ffcb680  
A2      : 0x3ffc2678  A3      : 0x00000000  A4      : 0x000000ff  A5      : 0x0000ff00  
A6      : 0x00ff0000  A7      : 0xff000000  A8      : 0x00000000  A9      : 0x3ffcb650  
A10     : 0x3ffc2678  A11     : 0x3f40065d  A12     : 0x00000000  A13     : 0x00000001  
A14     : 0x00000000  A15     : 0x3ffcb84c  SAR     : 0x0000000a  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000000  LBEG    : 0x400013f9  LEND    : 0x4000140d  LCOUNT  : 0xffffffff  

ELF file SHA256: 0000000000000000

Backtrace: 0x400013f9:0x3ffcb680 0x400d4b0b:0x3ffcb690 0x400d7a75:0x3ffcb880 0x4010169e:0x3ffcbd90 0x4009085e:0x3ffcbdb0

Rebooting...

**Comment by kbaggen** [comment]
Ok, I am pretty sure it is an issue on server side.

The ESP32 is going down at the time is suppose to call the "temperature control settings" and hence get a second json as:

{"settemp":"1","hysteresis":"0","tdevice":"9875142","temp_control_cooling":"0","temp_control_heating":"0","temp_control":"no","temp_control1":"no"}

So if we use other "username" and call this in browser : 
https://bubble-logger.com/get_iBLOPPER_temp_data.php?tdevice=9875142

You should get above json.

But if using your username it give nofthing and hence an error at logger side:
https://bubble-logger.com/get_iBLOPPER_temp_data.php?tdevice=9907448  

So I need to check this a bit as I guess it is a php sever side issue and not your board. But best advice right here and now would be to try with second board too.......!!!

**Reply by ESEMCE** [reply]
Well, I guess that is good news.
Unfortunately this is my only board at the moment..

As another option, can you delete my account, so I can see if it was just an issue caused in account setup?
I have just realised that one of the first problem solving changes that I made was to change my password to something that was far less complex. My initial password was long, complex and contained special characters ( generated by my password manager), the new one is still long, but far less complex.
I've just reset the password to something even simpler to no benefit.

**Reply by kbaggen** [reply]
I made some update to the php code and now it works in bowser, and by the way you have a test brew running now with 9907448_test.

So now I get by running browser: https://bubble-logger.com/get_iBLOPPER_temp_data.php?tdevice=9907448

{"settemp":"0","hysteresis":"0","tdevice":"9907448","temp_control_cooling":"0","temp_control_heating":"0","temp_control":"no","temp_control1":"no"}

So try test it again when you got the chance.

**Reply by ESEMCE** [reply]
I think it might be working!
I just have it powered on, no PC logs.
But it is no longer rebooting and if I tap the desk, the blue light flashes.

**Reply by kbaggen** [reply]
Should be working then :-) 
I at least can see your SPINDELMATE in the database now.
Thanks for holding out so far and solving an issue potentially impacting many users!

**Reply by ESEMCE** [reply]
Appreciate your work, the beers are on me!
Now to pretty up this project.
It's part of a present for my wife who is the brewer in our house, I'm merely the customer! :D

**Comment by ESEMCE** [comment]
Hell Klaus,
We has some success with the first brew, but although data was indeed coming through from the RAPT Pill, the Spindelmate page showed " Timestamp to old, Logger offline!!!"
I am now trying to integrate Temperature control with a Tasmota Athom V2 and Spindelmate says it can't find the smart plug.
The Smart Plug is definitely online (and on the same network) and I can control it from my PC.

I wonder if there is an account problem?

**Comment by kbaggen** [comment]
Hi agian, 
the data was comming fine into database.

But the entry screen of spindelmate did not.
I think the issue was the UTC time was set at 9.5 and need to be a sigle chiffer number as 9 or 10, so changed that, and I at least see it correctly now.

So yes, somehow a server side-issue or more an issue the php-coding do not check for this so to speak........
Take care and hopes on future good brews.
/Klaus

**Reply by ESEMCE** [reply]
One again thank you.
I tried adjusting the time offset during my last issues, but neglected to this time.
I do live in an area with a non standard time zone (we are in Summer Time now in Australia, so actually now +10.5 hours, but this is definitely something I can live with.

**Comment by kbaggen** [comment]
Oki, was not ware anyone had unequal time difference with ½ an hour. But I guess only a few is in this issue, so for now not something I gonna change.....

**Comment by ESEMCE** [comment]
You must hate me by now! :smile: 

It seems the scripting for tasmota temperature control is setting a PULSETIME of only 100 and as a result the switch is powering on for only around 10 seconds before turning off again.

Extract from my Tasmota console is below

13:45:53.426 HTP: Command
13:45:53.431 SRC: WebCommand from 192.168.3.85
13:45:53.434 CMD: Grp 0, Cmd 'PULSETIME', Idx 1, Len 3, Pld 100, Data '100'
13:45:53.437 RSL: RESULT = {"PulseTime1":{"Set":100,"Remaining":100}}
13:45:53.471 HTP: Command
13:45:53.476 SRC: WebCommand from 192.168.3.85
13:45:53.479 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 2, Pld 1, Data 'on'
13:45:53.483 RSL: RESULT = {"POWER":"ON"}
13:45:53.485 RSL: POWER = ON
13:45:53.575 CFG: Saved to flash at F4, Count 64, Bytes 4096
13:46:03.519 SRC: PulseTimer
13:46:03.523 RSL: RESULT = {"POWER":"OFF"}
13:46:03.525 RSL: POWER = OFF
13:46:03.589 CFG: Saved to flash at FB, Count 65, Bytes 4096
13:46:20.430 RSL: STATE = {"Time":"2023-10-30T13:46:20","Uptime":"0T00:59:43","UptimeSec":3583,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"WIFINAME REDACTED","BSSId":"MAC ID REDACTED","Channel":1,"Mode":"11n","RSSI":100,"Signal":-43,"LinkCount":1,"Downtime":"0T00:00:05"}}
13:46:20.441 RSL: SENSOR = {"Time":"2023-10-30T13:46:20","ENERGY":{"TotalStartTime":"2023-10-30T12:46:44","Total":0.005,"Yesterday":0.000,"Today":0.005,"Period":0,"Power":0,"ApparentPower":0,"ReactivePower":0,"Factor":0.00,"Voltage":0,"Current":0.000}}

**Reply by kbaggen** [reply]
From you settings you have 192.168.3.130 at both heating ang cooling, samewise both are set as off for now. So first of all, then I guess only play with one side for now.

Are you heating or cooling on that plug, as it makes it easier to solve........?
Both heating and cooling giving issues?
What plug you using?
What Tasmota firmware you using?


Just for info others has not reported it and I seem to not be able to redo it, so might also be you should ensure Tasmota plug is fully updated, etc?


My cooling side:

10:41:02.318 RSL: STATE = {"Time":"2023-11-03T10:41:02","Uptime":"4T17:35:45","UptimeSec":408945,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"HomeBox-C530_2.4G","BSSId":"C0:56:27:BE:B0:CA","Channel":2,"Mode":"11n","RSSI":88,"Signal":-56,"LinkCount":1,"Downtime":"0T00:00:07"}}
10:41:02.327 RSL: SENSOR = {"Time":"2023-11-03T10:41:02","ENERGY":{"TotalStartTime":"2023-10-29T17:05:25","Total":1.092,"Yesterday":0.170,"Today":0.000,"Period": 0,"Power": 0,"ApparentPower": 0,"ReactivePower": 0,"Factor":0.00,"Voltage":233,"Current":0.000}}
10:44:45.124 RSL: RESULT = {"PulseTime1":{"Set":490,"Remaining":490}}
10:44:45.156 RSL: RESULT = {"POWER":"ON"}
10:44:45.159 RSL: POWER = ON


And heating side:

11:06:02.349 RSL: STATE = {"Time":"2023-11-03T11:06:02","Uptime":"4T18:00:45","UptimeSec":410445,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"HomeBox-C530_2.4G","BSSId":"C0:56:27:BE:B0:CA","Channel":2,"Mode":"11n","RSSI":68,"Signal":-66,"LinkCount":1,"Downtime":"0T00:00:07"}}
11:06:02.358 RSL: SENSOR = {"Time":"2023-11-03T11:06:02","ENERGY":{"TotalStartTime":"2023-10-29T17:05:25","Total":1.102,"Yesterday":0.170,"Today":0.010,"Period": 1,"Power": 0,"ApparentPower": 0,"ReactivePower": 0,"Factor":0.00,"Voltage":233,"Current":0.000}}
11:08:45.147 RSL: RESULT = {"PulseTime1":{"Set":490,"Remaining":490}}
11:08:45.178 RSL: RESULT = {"POWER":"ON"}
11:08:45.180 RSL: POWER = ON

**Comment by kbaggen** [comment]
BTW.....I am on [Tasmota 13.2.0 by Theo Arends](https://bit.ly/tasmota)

**Comment by ESEMCE** [comment]
Yes, confirming I am also on 13.2.0 and I have also tried to reset the device to defaults.
And also confirming I didn't have both heating and cooling enabled at the same time.
And finally confirming that without SPINDELMATE control, the Smart plug seems to operate as expected.
SPINDELMATE is running v1.5.0 and it's local IP is 192.168.3.85

TLDR:
I'm using SPINDELMATE temperature control in reverse of your intention, but I think it should still work.

Ok this is going to be complicated. It took me a bit of time to get my head around it.

We have an immersion chiller system with it's own temperature control, but it is unreliable. Power can be on, but the chiller is no longer chilling even when it the temp is higher than set. Power cycling the chiller can resolve the issue, but if we are at work, we can't control the power.

I started testing on heating, with the SPINDELMATE set temperate just higher than the target for the chiller.

In my mind this should make the SPINDELMATE keep power ON to achieve the "heating" to get to the set temp.
If the chiller fails, the set temp will be reached by fermentation activity and SPINDELMATE would power off the chiller Automatically.
When I notice, I could remotely alter SPINDELMATE settings to a higher set temperature which would toggle on the chiller again (hopefully chilling again) and therefore bringing the brew back down to the actual target brew temp.

**Reply by ESEMCE** [reply]
Your SPINDELMATE is sending a Pulsetimer of only 490? So 49 seconds only?
My understanding was that it should be the SPINDELMATE cycle time (ie minimum 3 minutes) +60 seconds, so 2400 if the cycle time is set to 3 minutes?

**Reply by kbaggen** [reply]
UPDATED se below!

**Comment by kbaggen** [comment]
I changed a bit in the setting behind, so give me a hit if it help.........! So hopefully it give differently than 100.......e.g. 310 for 3min....

For 6min you should get 520 as the calculaton is 100+mins multiplied  60+60s, eg.:
3min =  100+3 multiplied 60+60 = 310
6min = 100 + 6 multiplied 60+60 = 520 

The 490 I hold is the version I am running is during only 100 + 6*60 + 30. I use my iBLOPPER project for this current brew I got running, but it is same code in regards of tempcontrol...... :-) 

BTW.....what is you plug, name model etc?

**Reply by ESEMCE** [reply]
Another problem resolved.
Thanks Klaus

PS. The smart plug I am using is Tasmota Athom V2 with an Australian pin layout (I imagine it is the same internals as the EU version as we have the same voltage)
https://www.athom.tech/blank-1/au-plug

**Reply by kbaggen** [reply]
No worries, it is my who thanks as this is how errors get ironed out.

Spread the word :-)
